e2e-tests/playwright/utils/authenticationProviders/helper.ts

```diff
@@ -0,0 +1,320 @@
+import { logger } from './Logger';
+import { exec } from 'child_process';
+import * as constants from './constants';
+import { expect } from '@playwright/test';
+import { kubeCLient } from './k8sHelper';
+import { V1ConfigMap, V1Secret } from '@kubernetes/client-node';
+
+export const k8sClient = new kubeCLient();
+
+export async function runShellCmd(command: string) {
+  return new Promise<string>((resolve, reject) => {
+    logger.info(`Executing command ${command}`);
+    const process = exec(command);
+    let result: string;
+    process.stdout.on('data', data => {
+      result = data;
+    });
+    process.stderr.on('data', data => {
+      logger.log({
+        level: 'error',
+        message: `Error occurred while running command`,
+        dump: data,
+      });
+      reject();
+      throw Error('Error executing shell command');
+    });
+    process.on('exit', () => resolve(result));
+  });
+}
+
+export async function upgradeHelmChartWithWait(
+  RELEASE: string,
+  CHART: string,
+  NAMESPACE: string,
+  VALUES: string,
+  CHART_VERSION: string,
+  QUAY_REPO: string,
+  TAG_NAME: string,
+  FLAGS: Array<string>,
+) {
+  logger.info(`Deleting any exisitng helm release ${RELEASE}`);
+  await deleteHelmReleaseWithWait(RELEASE, NAMESPACE);
```

Comment:
Is there a reason why you need to delete this first?
`helm upgrade` should be able to handle all changes in deployment without deleting release first

Reply:
Some of the test cases are applying changes at runtime to the authentication providers and checking the sync process and its results: the helm upgrade doesn't clear the data previously synced, unless you delete the postgres database, and that lead to inconsistencies in some tests, especially when they were retried for some reason. 
It also helps ensuring that each test case can be retried, mitigating any flakiness caused by the target cluster we are testing against. 
I guess there are different ways to handle this, but this was the simplest one that would also reduce the execution time of the whole suite; deleting and re-creating straight away ensures I can only wait for the first sync to happen, for example, without having to estimate when the next iteration will be or waiting for a longer time than needed.